### PR TITLE
Build emacsGit without Xcode

### LINKED ISSUE #286: Is Xcode required for emacsGit on Darwin?
I'm trying to build emacsGit without an xcode installation on aarch64-darwin:

```console
$ nix build -L .#emacsGit
warning: Git tree '/private/tmp/emacs-overlay' is dirty
tree-sitter-grammars> xcode-select: error: no developer tools were found at '/Applications/Xcode.app', and no install could be requested (perhaps no UI is present), please install manually from 'developer.apple.com'.
error: builder for '/nix/store/ns9xz4xa545g6dcsmcwlwp0g9whfni01-tree-sitter-grammars.drv' failed with exit code 1;
```

It comes from the `/usr/bin/install_name_tool` call: https://github.com/nix-community/emacs-overlay/blob/a201dcf3d712cf6d3934b396d523041781ff9589/overlays/emacs.nix#L86-L88

Is it expected that we should install Xcode, or does this show that solution is "not kosher"?  Thanks!

<details>
<summary>
System info
</summary>

```console
$ nix-shell -p nix-info --run "nix-info -m"
 - system: `"aarch64-darwin"`
 - host os: `Darwin 22.2.0, macOS 13.1`
 - multi-user?: `yes`
 - sandbox: `no`
 - version: `nix-env (Nix) 2.12.0`
 - channels(root): `"nixpkgs"`
 - nixpkgs: `/nix/var/nix/profiles/per-user/root/channels/nixpkgs`
```
</details>

### PR DESCRIPTION
**Pull request by rossabaker**
Fixes #286.

### CODE REVIEW

File: overlays/emacs.nix [hunk 1]
```diff
@@ -108,14 +109,14 @@ let
               tree-sitter-typescript
               tree-sitter-yaml
             ];
-            tree-sitter-grammars = super.runCommand "tree-sitter-grammars" {}
+            tree-sitter-grammars = super.runCommandCC "tree-sitter-grammars" {}
```
Comment (rossabaker) [inline overlays/emacs.nix:112]: `runCommandCC` exposes a Nix-managed `install_name_tool` to the command.

File: overlays/emacs.nix [hunk 2]
```diff
@@ -82,9 +82,10 @@ let
             lib = drv: ''lib${libName drv}.${libSuffix}'';
             linkCmd = drv:
               if super.stdenv.isDarwin
-              then ''cp ${drv}/parser $out/lib/${lib drv}
-                     # FIXME: Is this kosher?
-                     /usr/bin/install_name_tool -id $out/lib/${lib drv} $out/lib/${lib drv}
+              then ''cp ${drv}/parser .
+                     chmod +w ./parser
+                     install_name_tool -id $out/lib/${lib drv} ./parser
+                     cp ./parser $out/lib/${lib drv}
```
Comment (rossabaker) [inline overlays/emacs.nix:88]: Running just `install_name_tool` ran into a permissions error rewriting from the Nix store.  I needed to copy each parser  to the working directory, run the tool there, and _then_ copy to `$out`.

File: overlays/emacs.nix [hunk 3]
```diff
@@ -82,9 +82,10 @@ let
...
                      /usr/bin/codesign -s - -f $out/lib/${lib drv}
```
Comment (rossabaker) [inline overlays/emacs.nix:89]: This is another dubious `/usr/bin` reference, but unlike `install_name_tool`, doesn't need Xcode.

**Review by rossabaker — COMMENTED** [review]
I have no idea what I'm doing, but this built and I could run treesit on a bash file.

**Comment by leungbk** [comment]
Thanks!

**Pull request merged**